packages/flutter/test/material/date_range_picker_test.dart

```diff
@@ -122,6 +122,90 @@ void main() {
     });
   });
 
+  testWidgets('Last month header should be visible if last date is selected',
+      (WidgetTester tester) async {
+    firstDate = DateTime(2015, DateTime.january, 1);
+    lastDate = DateTime(2016, DateTime.december, 31);
+    initialDateRange = DateTimeRange(
+      start: lastDate,
+      end: lastDate,
+    );
+    await preparePicker(tester, (Future<DateTimeRange> range) async {
+      // December header should be showing, but no November
+      expect(find.text('December 2016'), findsOneWidget);
+      expect(find.text('November 2016'), findsNothing);
+    });
+  });
+
+  testWidgets('First month header should be visible if first date is selected',
+      (WidgetTester tester) async {
+    firstDate = DateTime(2015, DateTime.january, 1);
+    lastDate = DateTime(2016, DateTime.december, 31);
+    initialDateRange = DateTimeRange(
+      start: firstDate,
+      end: firstDate,
+    );
+    await preparePicker(tester, (Future<DateTimeRange> range) async {
+      // January and February headers should be showing, but no March
+      expect(find.text('January 2015'), findsOneWidget);
+      expect(find.text('February 2015'), findsOneWidget);
+      expect(find.text('March 2015'), findsNothing);
+    });
+  });
+
+  testWidgets('Current month header should be visible if no date is selected',
+      (WidgetTester tester) async {
+    firstDate = DateTime(DateTime.now().year - 1);
+    lastDate = DateTime(DateTime.now().year + 1);
+    initialDateRange = null;
+
+    String _getMonthName(int month) {
+      switch (month) {
+        case DateTime.january:
+          return 'January';
+        case DateTime.february:
+          return 'February';
+        case DateTime.march:
+          return 'March';
+        case DateTime.april:
+          return 'April';
+        case DateTime.may:
+          return 'May';
+        case DateTime.june:
+          return 'June';
+        case DateTime.july:
+          return 'July';
+        case DateTime.august:
+          return 'August';
+        case DateTime.september:
+          return 'September';
+        case DateTime.october:
+          return 'October';
+        case DateTime.november:
+          return 'November';
+        case DateTime.december:
+          return 'December';
+      }
+      return '';
+    }
+
+    await preparePicker(tester, (Future<DateTimeRange> range) async {
+      final DateTime currentMonth = DateTime.now();
```

Comment:
It is not a good practice to have tests dependent on things like the current date, as the results could vary between runs. A better practice here would be to just set the `currentDate` to a known constant and check against that. That way you wouldn't need to compute the month names as you would be testing against the one that you set.

Reply:
I was not aware that `DateRangePicker` allows `currentDate` to be set, haha. Thank you for the best practice advice!